src/tennis/core.clj

```diff
@@ -0,0 +1,131 @@
+(ns tennis.core)
+
+(def config {:max-matches 3
+             :labels [0 15 30 40 "A"]})
+
+
+(def default-scores {:player1 [0 [0] 0]
+                     :player2 [0 [0] 0]})
+
+(def scores (atom default-scores))
+
+
+(defn inc-game-only?
+  "when winning a game increases only the game score"
+  [game game-adv]
+  (or (< game 3)
+      (= game game-adv)))
+
+
+(defn inc-set-match?
+  "when winning a game leads to increase set and match scores"
+  [game game-adv set set-adv]
+  (and (>= (- game game-adv) 1)
+       (>= set 5)
+       (>= (- set set-adv) 1)))
+
+
+(defn inc-set-only?
+  "when winning a game increases only set score"
+  [game game-adv set set-adv]
+  (and (>= game 3)
+       (>= (- game game-adv) 1)
+       (or (< set 6)
+           (= (- set set-adv) -1)
+           (= (- set set-adv) 0))))
+
+
+(defn adversary-advantage?
+  "when adversary has advantage, then it defaults to 40-40(deuce) for both players"
+  [game game-adv]
+  (= (- game game-adv) -1))
+
+
+(defn increase-score
```

Comment:
:+1:  I like the idea to have this function with two sets of arguments:
 - with one argument - convenient to serve the game,
 - with two arguments - useful to test specific state change. 

"Doc string" would be welcomed here.

Reply:
Thank you! I added the doc string as suggested ;)